From 063e65397a89e3c7a71d388c5928366ad8f369b9 Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Mon, 28 Mar 2016 20:37:17 +0800
Subject: [PATCH] ARM64: dts: rk3399-tb: fix up the pwm regulator node

add pwm init voltage and id for uboot.
fix up the pwms node and add pwm polarity.

Change-Id: I4159c97ae498411ab958c2b1e1223139ac670452
Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-monkey.dts | 10 ++++++++++
 arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi    |  6 ++++--
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-monkey.dts b/arch/arm64/boot/dts/rockchip/rk3399-monkey.dts
index 0225b200001e..7774cc030184 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-monkey.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-monkey.dts
@@ -130,6 +130,16 @@
 	};
 };
 
+&vdd_log {
+	rockchip,pwm_id= <2>;
+	rockchip,pwm_voltage = <900000>;
+};
+
+&vdd_center {
+	rockchip,pwm_id= <3>;
+	rockchip,pwm_voltage = <900000>;
+};
+
 &fb {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi
index 949e1dd61b60..d21bc61eb4d0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi
@@ -48,21 +48,23 @@
 
 	vdd_log: vdd-log {
 		compatible = "pwm-regulator";
-		pwms = <&pwm2 0 1000>;
+		pwms = <&pwm2 0 25000 0>;
 		regulator-name = "vdd_log";
 		regulator-min-microvolt = <800000>;
 		regulator-max-microvolt = <1400000>;
 		regulator-always-on;
+		regulator-boot-on;
 		status = "okay";
 	};
 
 	vdd_center: vdd-center {
 		compatible = "pwm-regulator";
-		pwms = <&pwm3 0 1000>;
+		pwms = <&pwm3 0 25000 0>;
 		regulator-name = "vdd_center";
 		regulator-min-microvolt = <800000>;
 		regulator-max-microvolt = <1400000>;
 		regulator-always-on;
+		regulator-boot-on;
 		status = "okay";
 	};
 
-- 
2.35.3

